kube/Cargo.toml

```
@@ -16,57 +16,57 @@ categories = ["web-programming::http-client"]
 edition = "2018"
 
 [features]
-default = ["native-tls"]
+default = ["native-tls", "client"]
 native-tls = ["openssl", "hyper-tls", "tokio-native-tls"]
 rustls-tls = ["hyper-rustls", "tokio-rustls"]
```

```suggestion
default = ["native-tls"]
native-tls = ["client", "openssl", "hyper-tls", "tokio-native-tls"]
rustls-tls = ["client", "hyper-rustls", "tokio-rustls"]
```
I think this works.
We can also add `client` to other features that doesn't make sense without `client` like `ws`, `oauth`, `gzip`.

---

this does introduce having features depend on features also, which will make them slightly harder to follow if it gets heavily used. it probably isn't a problem either way, put a static assert for this case in lib.

---

Thinking more about it, this does seem like a better way to declare feature dependencies than with assertions. Will continue tomorrow

---

Yeah, features _should_ be additive, but some features depends on another implicitly already. I think it's better to make that explicit.
Not sure how common, but I've seen this pattern used in some crates I've used.

---

Have added `client` to most client-using feature sets and removed the static assert.

Also separated the dependencies into `client`, `__config` and `__non_core` in the hopes that we can make a proper `config` feature (by checking of config or __non_core).